embulk-deps/guess/src/main/java/org/embulk/deps/guess/CharsetDetectorImpl.java

```diff
@@ -0,0 +1,22 @@
+package org.embulk.deps.guess;
+
+import com.ibm.icu.text.CharsetDetector;
+
+public final class CharsetDetectorImpl extends org.embulk.deps.guess.CharsetDetector {
+    public CharsetDetectorImpl() {
+        this.detector = new CharsetDetector();
+    }
+
+    @Override
+    public CharsetDetectorImpl setText(final byte[] in) {
+        this.detector.setText(in);
+        return this;
+    }
+
+    @Override
+    public CharsetMatchImpl detect() {
+        return new CharsetMatchImpl(this.detector.detect());
```

Comment:
Not quite sure, but won't it delegate the loading request to the parent class loader?

Reply:
If I'm not making mistakes...
* This `CharsetMatchImpl` is `org.embulk.deps.guess. CharsetMatchImpl ` defined in the same package `embulk-deps-guess` (just below) loaded by the sub ClassLoader.
* The `detector` instance is `com.ibm.icu.text.CharsetDetector` (as in `import`), which is ICU4J's also loaded by the sub ClassLoader.
* Then, `detector.detect()` returns `com.ibm.icu.text.CharsetMatch`, which is in ICU4J again loaded by the sub ClassLoader, and that is used to construct `org.embulk.deps.CharsetMatchImpl`.

So, I believe it completes in the sub ClassLoader, but if you still see points in question, please let me know.

Reply:
forgot it's in ` embulk-deps/guess`